libyul/Object.cpp

```diff
@@ -65,9 +65,45 @@ set<YulString> Object::dataNames() const
 {
 	set<YulString> names;
 	names.insert(name);
-	for (auto const& subObject: subIndexByName)
-		names.insert(subObject.first);
-	// The empty name is not valid
```

Comment:
https://github.com/ethereum/solidity/pull/9047#discussion_r431780243

This comment applies to `names.erase(YulString{});` that's still present below and it's pretty informative so I think it should stay.